tests/src/testfebases.cpp

```diff
@@ -0,0 +1,108 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include <config.h>
+
+#include "testhelpers.hh"
+
+#include <dune/common/test/testsuite.hh>
+#include <dune/functions/functionspacebases/basistags.hh>
+#include <dune/functions/functionspacebases/compositebasis.hh>
+#include <dune/functions/functionspacebases/lagrangebasis.hh>
+#include <dune/functions/functionspacebases/powerbasis.hh>
+#include <dune/grid/yaspgrid.hh>
+
+#include <ikarus/finiteelements/febases.hh>
+
+using Dune::TestSuite;
+
+#include <ikarus/utils/basis.hh>
+#include <ikarus/utils/init.hh>
+
+template <typename GridView, typename PreBasis>
+auto getBasis(const GridView& gridView, const PreBasis& preBasis) {
+  const auto basis      = Ikarus::makeBasis(gridView, preBasis);
+  const auto& localView = basis.flat().localView();
+  auto element          = elements(gridView).begin();
+  return Ikarus::FEBases<decltype(basis)>(basis, *element);
+}
+
+template <typename TestSuiteType, typename... Bases>
+void checkBases(TestSuiteType& t, const std::vector<int>& expectedNumberOfChildren,
+                const std::vector<int>& expectedSize, const std::string& basisName, const Bases&... bases) {
```

Comment:
You know the size at compile time why use the runtime assert to checke the size of expectedSize. I would suggest creating a std::array